tests/build_tree_test.py

```diff
@@ -345,206 +346,217 @@ def test_check_recipe_path_package_field():
-                                                    python="2.6",
-                                                    build_type="cuda",
-                                                    mpi_type="openmpi",
-                                                    build_command_dependencies=[],
-                                                    run_dependencies=[],
-                                                    build_dependencies=[],
-                                                    host_dependencies=[],
-                                                    test_dependencies=[]),
-
-                                build_tree.BuildCommand("recipe3",
-                                                    "repo3",
-                                                    ["package3a"],
-                                                    output_files=["linux/package3a.tar.gz"],
-                                                    python="2.6",
-                                                    build_type="cpu",
-                                                    mpi_type="openmpi",
-                                                    build_command_dependencies=[1],
-                                                    run_dependencies=[],
-                                                    build_dependencies=["package1a"],
-                                                    host_dependencies=[],
-                                                    test_dependencies=[])]
-    additional_build_commands = [build_tree.BuildCommand("recipe4",
-                                                    "repo4",
-                                                    ["package4a"],
-                                                    output_files=["linux/package4a.tar.gz"],
-                                                    python="2.6",
-                                                    build_type="cpu",
-                                                    mpi_type="openmpi",
-                                                    build_command_dependencies=[],
-                                                    run_dependencies=[],
-                                                    build_dependencies=[],
-                                                    host_dependencies=[],
-                                                    test_dependencies=[])]
-
-    out_commands, _ = build_tree._add_build_command_dependencies(additional_build_commands,initial_build_commands,len(initial_build_commands))
-    assert len(out_commands)==1  # Make sure the non-duplicates are not removed
-
-    out_commands, _ = build_tree._add_build_command_dependencies(duplicate_build_commands, initial_build_commands, len(initial_build_commands))
-    assert len(out_commands)==1
-
-    for build_command in out_commands:
-        assert build_command.build_command_dependencies == [0]
+# def test_build_tree_cycle_fail():
```

Comment:
Can we still leave in some sort of test that makes sure it handles cycles correctly? Or maybe that is already happening somewhere else since python introduces cycles.

Reply:
Yeah, that's a good idea. I'll add a test along those lines.

Reply:
I added cycle detection back, but it only checks packages being built.